File: snac_record.ts
import { verifyIdentifier } from "./utility.ts";
import { normalizeSNAC, validateSNAC } from "./snac.ts";

/**
 * Verifies an SNAC ID using an SNAC Cooperative website. Identifiers are normalized before
 * verifying.
 * @param id, the SNAC identifier
 * @returns Promise<boolean> - True if the SNAC ID exists, otherwise false.
 */
export async function verifySNAC(id: string): Promise<boolean> {
  const normalizedID = normalizeSNAC(id);
  return await verifyIdentifier(
    id,
    `https://snaccooperative.org/view/${normalizedID}`,
    validateSNAC,
  );
}

File: snac_record.go
package metadatatools

// VerifySNAC verifies a SNAC ID exists via the SNAC Cooperative website.
// The identifier is normalized before verifying.
func VerifySNAC(id string) bool {
	normalized := NormalizeSNAC(id)
	return VerifyIdentifier(id, "https://snaccooperative.org/view/"+normalized, ValidateSNAC)
}
